### ISSUE TITLE
Sphinx parallel build error (4.3.1 regression)

### ISSUE THREAD

**Issue by pitrou** [sphinx-doc/sphinx#9912]
### Describe the bug

After upgrading to Sphinx 4.3.1, I get the following error when building my project:
```
[...]
Recherche des fichiers périmés... aucun résultat
Environnement de sérialisation... fait
Vérification de la cohérence... fait
Document en préparation... fait
Écriture... [ 21%] python/generated/pyarrow.BufferReader .. python/generated/pyarrow.HadoopFileSystem.chmod                                                                 
Sphinx parallel build error:
AttributeError: can't set attribute
make: *** [Makefile:76 : html] Erreur 2
```

(I removed generated files using `git clean -Xfd .` just in case)

Downgrading to 4.2.0, the build works again.

### How to Reproduce

Unfortunately not easy as some C++ build steps are involved :-( But I'm happy to try any diagnosis you might suggest.

### Your project

https://github.com/apache/arrow/tree/master/docs

### OS

Ubuntu 20.04

### Python version

3.9.7

### Sphinx version

4.3.1

### Sphinx extensions

```python
extensions = [
    'sphinx.ext.autodoc',
    'sphinx.ext.autosummary',
    'sphinx.ext.doctest',
    'sphinx.ext.ifconfig',
    'sphinx.ext.mathjax',
    'sphinx.ext.viewcode',
    'sphinx.ext.napoleon',
    'IPython.sphinxext.ipython_directive',
    'IPython.sphinxext.ipython_console_highlighting',
    'breathe',
    'sphinx_tabs.tabs'
]
```

**Comment by tk0miya** [comment]
Could you try building your document in non-parallel mode, please? Then we would be able to see the detail of the error.

**Comment by pitrou** [comment]
Ok, here is the error:
```
# Sphinx version: 4.3.1
# Python version: 3.9.7 (CPython)
# Docutils version: 0.16 release
# Jinja2 version: 3.0.2
# Last messages:
#   Écriture... [  2%] cpp/api/ipc
#   Écriture... [  2%] cpp/api/memory
#   Écriture... [  2%] cpp/api/scalar
#   Écriture... [  2%] cpp/api/support
#   Écriture... [  2%] cpp/api/table
#   Écriture... [  2%] cpp/api/tensor
#   Écriture... [  2%] cpp/api/utilities
#   Écriture... [  3%] cpp/arrays
#   Écriture... [  3%] cpp/build_system
#   Écriture... [  3%] cpp/compute
# Loaded extensions:
#   sphinx.ext.mathjax (4.3.1) from /home/antoine/miniconda3/envs/pyarrow/lib/python3.9/site-packages/sphinx/ext/mathjax.py
#   sphinxcontrib.applehelp (1.0.2) from /home/antoine/miniconda3/envs/pyarrow/lib/python3.9/site-packages/sphinxcontrib/applehelp/__init__.py
#   sphinxcontrib.devhelp (1.0.2) from /home/antoine/miniconda3/envs/pyarrow/lib/python3.9/site-packages/sphinxcontrib/devhelp/__init__.py
#   sphinxcontrib.htmlhelp (2.0.0) from /home/antoine/miniconda3/envs/pyarrow/lib/python3.9/site-packages/sphinxcontrib/htmlhelp/__init__.py
#   sphinxcontrib.serializinghtml (1.1.5) from /home/antoine/miniconda3/envs/pyarrow/lib/python3.9/site-packages/sphinxcontrib/serializinghtml/__init__.py
#   sphinxcontrib.qthelp (1.0.3) from /home/antoine/miniconda3/envs/pyarrow/lib/python3.9/site-packages/sphinxcontrib/qthelp/__init__.py
#   alabaster (0.7.12) from /home/antoine/miniconda3/envs/pyarrow/lib/python3.9/site-packages/alabaster/__init__.py
#   sphinx.ext.autodoc.preserve_defaults (1.0) from /home/antoine/miniconda3/envs/pyarrow/lib/python3.9/site-packages/sphinx/ext/autodoc/preserve_defaults.py
#   sphinx.ext.autodoc.type_comment (4.3.1) from /home/antoine/miniconda3/envs/pyarrow/lib/python3.9/site-packages/sphinx/ext/autodoc/type_comment.py
#   sphinx.ext.autodoc (4.3.1) from /home/antoine/miniconda3/envs/pyarrow/lib/python3.9/site-packages/sphinx/ext/autodoc/__init__.py
#   sphinx.ext.autosummary (4.3.1) from /home/antoine/miniconda3/envs/pyarrow/lib/python3.9/site-packages/sphinx/ext/autosummary/__init__.py
#   sphinx.ext.doctest (4.3.1) from /home/antoine/miniconda3/envs/pyarrow/lib/python3.9/site-packages/sphinx/ext/doctest.py
#   sphinx.ext.ifconfig (4.3.1) from /home/antoine/miniconda3/envs/pyarrow/lib/python3.9/site-packages/sphinx/ext/ifconfig.py
#   sphinx.ext.viewcode (4.3.1) from /home/antoine/miniconda3/envs/pyarrow/lib/python3.9/site-packages/sphinx/ext/viewcode.py
#   sphinx.ext.napoleon (4.3.1) from /home/antoine/miniconda3/envs/pyarrow/lib/python3.9/site-packages/sphinx/ext/napoleon/__init__.py
#   IPython.sphinxext.ipython_directive (unknown version) from /home/antoine/miniconda3/envs/pyarrow/lib/python3.9/site-packages/IPython/sphinxext/ipython_directive.py
#   IPython.sphinxext.ipython_console_highlighting (unknown version) from /home/antoine/miniconda3/envs/pyarrow/lib/python3.9/site-packages/IPython/sphinxext/ipython_console_highlighting.py
#   breathe (4.31.0) from /home/antoine/miniconda3/envs/pyarrow/lib/python3.9/site-packages/breathe/__init__.py
#   sphinx_tabs.tabs (unknown version) from /home/antoine/miniconda3/envs/pyarrow/lib/python3.9/site-packages/sphinx_tabs/tabs.py
#   pydata_sphinx_theme (unknown version) from /home/antoine/miniconda3/envs/pyarrow/lib/python3.9/site-packages/pydata_sphinx_theme/__init__.py
Traceback (most recent call last):
  File "/home/antoine/miniconda3/envs/pyarrow/lib/python3.9/site-packages/sphinx/cmd/build.py", line 280, in build_main
    app.build(args.force_all, filenames)
  File "/home/antoine/miniconda3/envs/pyarrow/lib/python3.9/site-packages/sphinx/application.py", line 344, in build
    self.builder.build_update()
  File "/home/antoine/miniconda3/envs/pyarrow/lib/python3.9/site-packages/sphinx/builders/__init__.py", line 294, in build_update
    self.build(to_build,
  File "/home/antoine/miniconda3/envs/pyarrow/lib/python3.9/site-packages/sphinx/builders/__init__.py", line 358, in build
    self.write(docnames, list(updated_docnames), method)
  File "/home/antoine/miniconda3/envs/pyarrow/lib/python3.9/site-packages/sphinx/builders/__init__.py", line 532, in write
    self._write_serial(sorted(docnames))
  File "/home/antoine/miniconda3/envs/pyarrow/lib/python3.9/site-packages/sphinx/builders/__init__.py", line 542, in _write_serial
    self.write_doc(docname, doctree)
  File "/home/antoine/miniconda3/envs/pyarrow/lib/python3.9/site-packages/sphinx/builders/html/__init__.py", line 626, in write_doc
    self.docwriter.write(doctree, destination)
  File "/home/antoine/miniconda3/envs/pyarrow/lib/python3.9/site-packages/docutils/writers/__init__.py", line 78, in write
    self.translate()
  File "/home/antoine/miniconda3/envs/pyarrow/lib/python3.9/site-packages/sphinx/writers/html.py", line 71, in translate
    self.document.walkabout(visitor)
  File "/home/antoine/miniconda3/envs/pyarrow/lib/python3.9/site-packages/docutils/nodes.py", line 214, in walkabout
    if child.walkabout(visitor):
  File "/home/antoine/miniconda3/envs/pyarrow/lib/python3.9/site-packages/docutils/nodes.py", line 214, in walkabout
    if child.walkabout(visitor):
  File "/home/antoine/miniconda3/envs/pyarrow/lib/python3.9/site-packages/docutils/nodes.py", line 214, in walkabout
    if child.walkabout(visitor):
  [Previous line repeated 1 more time]
  File "/home/antoine/miniconda3/envs/pyarrow/lib/python3.9/site-packages/docutils/nodes.py", line 206, in walkabout
    visitor.dispatch_visit(self)
  File "/home/antoine/miniconda3/envs/pyarrow/lib/python3.9/site-packages/sphinx/util/docutils.py", line 477, in dispatch_visit
    method(node)
  File "/home/antoine/miniconda3/envs/pyarrow/lib/python3.9/site-packages/pydata_sphinx_theme/bootstrap_html_translator.py", line 40, in visit_table
    self._table_row_index = 0
AttributeError: can't set attribute
```

**Comment by tk0miya** [comment]
Sorry for late response. This was a bug(?) of the pydata-sphinx-theme and it was already fixed in v0.7.2 (see https://github.com/pydata/pydata-sphinx-theme/issues/508).
Please try it.

Thanks,

**Comment by pitrou** [comment]
Indeed, it worked, thank you!
